app/controllers/users/admin_roles.js

```diff
@@ -0,0 +1,154 @@
+import Ember from "ember";
+import _ from "lodash";
+
+export default Ember.Controller.extend({
+  printerService: Ember.inject.service(),
+  apiBaseService: Ember.inject.service(),
+  userService: Ember.inject.service(),
+
+  selectedPrinterId: "",
+  user: Ember.computed.alias("model.user"),
+  noAdminAppAccess: Ember.computed.equal("adminRoleAccess", "noAccess"),
+
+  printers: Ember.computed(function() {
+    return this.get("printerService").allAvailablePrinters();
+  }),
+
+  canUpdateRole: Ember.computed("user.id", function() {
+    return this.get("userService").canUpdateRole(this.get("user.id"));
+  }),
+
+  selectedPrinterDisplay: Ember.computed(
+    "model.user.id",
+    "selectedPrinterId",
+    function() {
+      const printerId = this.get("selectedPrinterId");
+      return this.get("userService").getPrinterForUser(
+        this.get("user"),
+        printerId,
+        "admin"
+      );
+    }
+  ),
+
+  roleError: Ember.computed("noAdminAppRole", "adminRoleAccess", function() {
+    return (
+      this.get("noAdminAppRole") && this.get("adminRoleAccess") !== "noAccess"
+    );
+  }),
+
+  adminRoleAccess: Ember.computed("user.roles.[]", {
+    get() {
+      if (this.get("noAdminAppRole")) {
+        return "noAccess";
```

Comment:
rather than using strings, I'd recommend creating an enum

e.g

```javascript
const ACCESS_TYPES = {
     NO_ACCESS:               'noAccess',
     LIMITED_ACCESS:       'accessTill',
     UNLIMITED_ACCESS: 'accessForever'
}
```